### PR TITLE
Update to go1.10.3

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: updates to build with go1.10.3; see list of changes [here](https://github.com/golang/go/issues?q=milestone%3AGo1.10.3).

We'll probably want to cherrypick this to release-1.11 as well.

**Release note**:

```release-note
Update to use go1.10.3
```

/assign @BenTheElder @cblecker

### CODE REVIEW

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -336,7 +336,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(go version)"
   local minimum_go_version
-  minimum_go_version=go1.10.2
+  minimum_go_version=go1.10.3
```
[the remark this answers is not in the document]
Comment (ixdy) [reply hack/lib/golang.sh]: I guess not. I can revert this, since it'll make CI easier to update.

**Comment by warmchang** [comment]
Hi, may I ask what is the policy to update the Go lang compiler? Keep following the latest version?

**Comment by ixdy** [comment]
@warmchang we generally try to keep up to date with the latest patch release (for example, 1.10.2 -> 1.10.3 here) for the relevant bug fixes. 

We update to the latest minor release (for example, 1.9 -> 1.10) less aggressively, since there are often bugs or other issues to be worked out first, and we want to ensure enough time to soak test the new release, so we don't update close to the kubernetes release date.

This is why kubernetes 1.11 is the first release to use go1.10: while go1.10 was released prior to kubernetes 1.10 being released, it was too close to our cut.

**Comment by warmchang** [comment]
Got it, thank you @ixdy !

**Pull request merged**